### PR TITLE
Update 

### PR DESCRIPTION
**Pull request by nwiltberg**
Hi @PeerHerholz, 
could you please check if what I'm doing here makes sense?
I don't really know how to proceed with the implementation of the algorithm.

Thanks!

### CODE REVIEW

File: code/Data_Analysis.ipynb [hunk 1]
```diff
@@ -1086,13 +1086,161 @@
...
+   ],
+   "source": [
+    "mean_scores"
+   ]
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "neuro_ai",
+   "display_name": "Python 3.9.13 64-bit (microsoft store)",
```
Comment (PeerHerholz) [inline code/Data_Analysis.ipynb]: Could you maybe ensure that you're using a dedicated `python environment` to run things? This might go a long way towards FAIR-ness and ease up the further project parts.

File: open_lab_notebook/calendar_week_48.md [hunk 2]
```diff
@@ -7,10 +7,12 @@
...
 
 ## Completed tasks
-- tried to implement supervised learning algorithm (1,5 hours)
+- tried to implement supervised learning algorithm (2 hours)
+- lecture about deep learning (1.5 hours)
+- looked through the c-pac outputs on brainlife (30 min.)
 
 ## Problems
-- 
+- don't know the difference between desc-1_regressors and desc-2_regressors
```
Comment (PeerHerholz) [inline open_lab_notebook/calendar_week_48.md:23]: I also don't know that at the top of my head, let's check it out on Friday.

File: code/Data_Analysis.ipynb [hunk 3]
```diff
@@ -1086,13 +1086,161 @@
     "connectome_vect=nilearn.connectome.sym_matrix_to_vec(correlation_matrices, discard_diagonal=True)\n",
     "connectome_vect"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ad1071bd",
+   "metadata": {},
+   "source": [
+    "## 4. Supervised learning algorithm"
```
Comment (PeerHerholz) [inline code/Data_Analysis.ipynb]: Could you maybe a few explanations (short points are ok) so that things can be evaluated respectively?

**Review by PeerHerholz — CHANGES_REQUESTED** [review]
Hi @nwiltberg,

thanks for updating, great stuff! I added a few comments, could you maybe have a look at them?

File: code/Data_Analysis.ipynb (hunk 1 above)
Comment (nwiltberg) [reply code/Data_Analysis.ipynb]: I don't even know what happened here to be honest. I wanted to use neuro_ai all along...

**Comment by nwiltberg** [comment]
Hi @PeerHerholz,
I adapted the part of the notebook about the supervised learning algorithm. Could you please check it?
Thx!

File: code/Data_Analysis.ipynb [hunk 4]
```diff
@@ -1096,143 +1096,285 @@
...
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 1 1 0] TEST: [0 3 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0] TEST: [0 1 1 0]\n"
+     ]
+    }
+   ],
    "source": [
-    "cv = StratifiedShuffleSplit(n_splits=30, random_state=0, test_size=10)"
+    "from sklearn.model_selection import KFold\n",
+    "kf = KFold(n_splits=10)\n",
```
Comment (PeerHerholz) [inline code/Data_Analysis.ipynb]: I think the prior CV was working fine and gave you the option to create more splits based on shuffling, which could be useful going forward.

File: code/Data_Analysis.ipynb [hunk 5]
```diff
@@ -1096,143 +1096,285 @@
...
-    "scores = []\n",
-    "\n",
-    "for train, test in cv.split(connectome_vect, groups):\n",
-    "    classifier = LinearSVC().fit(connectome_vect[train], groups[train])\n",
-    "    predictions = classifier.predict(connectome_vect[test])\n",
-    "    scores.append(accuracy_score(groups[test], predictions))"
+    "from sklearn.svm import LinearSVC\n",
+    "from sklearn.model_selection import cross_val_predict\n",
+    "est = LinearSVC()\n",
+    "pred = cross_val_predict(est, connectome_vect, groups, cv=10, method='predict')"
```
Comment (PeerHerholz) [inline code/Data_Analysis.ipynb]: if you set `cv=10` and not indicate the `cv` object you created above, a different `cv` will be applied. The `cross_val_predict` docs outline that if you set a number for `cv` it will run a StratifiedShuffleSplit (IIRC) but you can also set a previously defined `cv` here, e.g.

`my_cv = WhateverCVYouWantToUse`
`pred = cross_val_predict(est, connectome_vect, groups, cv=my_cv, method='predict')`

**Review by PeerHerholz — CHANGES_REQUESTED** [review]

**Review by PeerHerholz — CHANGES_REQUESTED** [review]
Hi @nwiltberg,

thx very much for this. Could you please push the latest version of the notebook as well, the one with the `error`?

Best, Peer

**Review by PeerHerholz — COMMENTED** [review]
What's the rationale for using both `cross_vaL_predict` and `cross_val_score`?
